Source/WTF/wtf/WeakHashSet.h

```diff
@@ -85,24 +88,16 @@ class WeakHashSet final {
     private:
         template <typename, typename, EnableWeakPtrThreadingAssertions> friend class WeakHashSet;
 
+        const WeakHashSet& m_set;
         typename WeakPtrImplSet::const_iterator m_position;
         typename WeakPtrImplSet::const_iterator m_endPosition;
     };
     typedef WeakHashSetConstIterator const_iterator;
 
     WeakHashSet() { }
 
-    const_iterator begin() const
-    {
-        increaseOperationCountSinceLastCleanup();
-        return WeakHashSetConstIterator(m_set, m_set.begin());
-    }
-
-    const_iterator end() const
-    {
-        increaseOperationCountSinceLastCleanup();
-        return WeakHashSetConstIterator(m_set, m_set.end());
-    }
+    const_iterator begin() const { return WeakHashSetConstIterator(*this, m_set.begin()); }
```

Comment:
We should probably drop the increaseOperationCountSinceLastCleanup() call in forEach() as well?